openapi/src/main/java/io/helidon/openapi/ExpandedTypeDescription.java

```diff
@@ -0,0 +1,303 @@
+     *     property description for {@code additionalProperties} that infers which variant a
+     *     specific node in the document actually uses and then processes it accordingly.
+     * </p>
+     */
+    static class SchemaTypeDescription extends ExpandedTypeDescription {
+
+        private static final PropertyDescriptor ADDL_PROPS_PROP_DESCRIPTOR = preparePropertyDescriptor();
+
+        private static final Property ADDL_PROPS_PROPERTY =
+                new MethodProperty(ADDL_PROPS_PROP_DESCRIPTOR) {
+
+                    @Override
+                    public void set(Object object, Object value) throws Exception {
+                        Schema s = Schema.class.cast(object);
+                        if (value instanceof Schema) {
+                            s.setAdditionalPropertiesSchema((Schema) value);
+                        } else {
+                            s.setAdditionalPropertiesBoolean((Boolean) value);
+                        }
+                    }
+
+                    @Override
+                    public Object get(Object object) {
+                        Schema s = Schema.class.cast(object);
+                        Boolean b = s.getAdditionalPropertiesBoolean();
+                        return b != null ? b : s.getAdditionalPropertiesSchema();
+                    }
+                };
+
+        private static PropertyDescriptor preparePropertyDescriptor() {
+            try {
+                return new PropertyDescriptor("additionalProperties",
+                        Schema.class.getMethod("getAdditionalPropertiesSchema"),
+                        Schema.class.getMethod("setAdditionalPropertiesSchema", Schema.class));
+            } catch (IntrospectionException | NoSuchMethodException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        private SchemaTypeDescription(Class<? extends Object> clazz, Class<?> impl) {
+            super(clazz, impl);
+        }
+
+        @Override
+        public Property getProperty(String name) {
+            return name.equals("additionalProperties") ? ADDL_PROPS_PROPERTY : super.getProperty(name);
+        }
+    }
+
+    /**
+     * Property description for an extension subnode.
+     */
+    static class ExtensionProperty extends Property {
+
+        ExtensionProperty(String name) {
+            super(name, Object.class);
+        }
+
+        @Override
+        public Class<?>[] getActualTypeArguments() {
+            return new Class[0];
+        }
+
+        @Override
+        public void set(Object object, Object value) throws Exception {
+            asExt(object).addExtension(getName(), value);
+        }
+
+        @Override
+        public Object get(Object object) {
+            return asExt(object).getExtensions().get(getName());
+        }
+
+        @Override
+        public List<Annotation> getAnnotations() {
+            return null;
```

Comment:
OK to return `null`, or should it be `Collections.emptyList()`?  Don't know the innards of SnakeYAML to know whether one is different from the other.  Obviously avoiding any object allocations is good if possible.

Reply:
Fixed.